src/main/java/com/microsoft/lst_bench/client/QueryResult.java

```diff
@@ -65,24 +68,30 @@ public Integer getValueListSize() {
   }
 
   public boolean containsEmptyResultColumnOnly() {
-    if (valueList.keySet().size() == 1
+    return valueList.keySet().size() == 1
         && valueList.containsKey(RESULT)
-        && valueList.get(RESULT).size() == 0) {
-      return true;
-    }
-    return false;
+        && valueList.get(RESULT).isEmpty();
   }
 
   public Map<String, Object> getStringMappings(int listMin, int listMax) {
     Map<String, Object> result = new HashMap<>();
-    for (String key : this.valueList.keySet()) {
+    for (String key : valueList.keySet()) {
       List<String> localList =
-          this.valueList.get(key).subList(listMin, listMax).stream()
-              .map(s -> s.toString())
+          valueList.get(key).subList(listMin, listMax).stream()
+              .map(Object::toString)
               .collect(Collectors.toUnmodifiableList());
-      // TODO: This assumes a VARCHAR type (or implicit casting by the engine),
-      //       we should probably handle it more generically using data types.
-      result.put(key, "'" + String.join("','", localList) + "'");
+      switch (columnTypes.get(key)) {
+        case java.sql.Types.BIGINT:
+        case java.sql.Types.INTEGER:
+        case java.sql.Types.SMALLINT:
+        case java.sql.Types.TINYINT:
+          result.put(key, String.join(",", localList));
+          break;
+        default:
```

Comment:
Interesting, is this an issue specific to Trino or why didn't we stumble across this before?

Reply:
Yes, it relates to the implicit casting rules for each system. Trino is stricter in the handling of data types, which I believe aligns more closely with the common approach across SQL systems than Spark's, so it made sense to adjust this logic to handle different data types.